File: SpringBootBackend/src/main/java/com/th/entity/Payment.java
package com.th.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Payment extends BaseEntity {
    @ManyToOne
    @JoinColumn(name = "booking_id", nullable = false)
    private Booking booking;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "vendor_id", nullable = false)
    private Vendor vendor;

    private String transactionId;
    private String paymentMethod;
    private BigDecimal amount;

    @Column(name = "payment_date", nullable = false)
    private LocalDateTime paymentDate;

    @Enumerated(EnumType.STRING)
    private PaymentStatus status = PaymentStatus.SUCCESS;

    public enum PaymentStatus {
        SUCCESS, FAILED, PENDING
    }
}

File: DOTNET BACKEND/TransitHive/Models/Payment.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Numerics;

namespace TransitHive.Models
{
    public class Payment
    {
        public int PaymentId { get; set; }
        public int BookingId { get; set; }
        public int UserId { get; set; }
        public int VendorId { get; set; }
        public string PaymentMethod { get; set; }
        public decimal Amount { get; set; }
        public DateTime PaymentDate { get; set; } = DateTime.Now;

        [Column(TypeName = "ENUM('SUCCESS', 'FAILED', 'PENDING')")]
        public PaymentStatus Status { get; set; } = PaymentStatus.SUCCESS;
        public Booking Booking { get; set; }
        public User User { get; set; }
        public Vendor Vendor { get; set; }
    }

    public enum PaymentStatus
    {
        SUCCESS, FAILED, PENDING
    }
}
